libs/parallelism/algorithms/include/hpx/parallel/algorithms/partial_sort.hpp

```diff
@@ -0,0 +1,537 @@
+    ///           type \a sequenced_task_policy or
+    ///           \a parallel_task_policy and returns void otherwise.
+    ///
+    template <typename RandIter>
+    typename util::detail::algorithm_result<ExPolicy>::type partial_sort(
+        ExPolicy&& policy, RandIter first, RandIter middle, RandIter last);
+
+}    // namespace hpx
+
+#else
+
+#include <hpx/config.hpp>
+#include <hpx/assert.hpp>
+#include <hpx/async_local/dataflow.hpp>
+#include <hpx/concepts/concepts.hpp>
+#include <hpx/functional/invoke.hpp>
+#include <hpx/functional/tag_invoke.hpp>
+#include <hpx/iterator_support/traits/is_iterator.hpp>
+#include <hpx/type_support/decay.hpp>
+
+#include <hpx/algorithms/traits/projected.hpp>
+#include <hpx/execution/algorithms/detail/predicates.hpp>
+#include <hpx/execution/executors/execution.hpp>
+#include <hpx/execution/executors/execution_information.hpp>
+#include <hpx/execution/executors/execution_parameters.hpp>
+#include <hpx/executors/exception_list.hpp>
+#include <hpx/executors/execution_policy.hpp>
+#include <hpx/parallel/algorithms/detail/dispatch.hpp>
+#include <hpx/parallel/algorithms/detail/is_sorted.hpp>
+#include <hpx/parallel/algorithms/sort.hpp>
+#include <hpx/parallel/util/compare_projected.hpp>
+#include <hpx/parallel/util/detail/algorithm_result.hpp>
+#include <hpx/parallel/util/detail/chunk_size.hpp>
+#include <hpx/parallel/util/projection_identity.hpp>
+
+#include <algorithm>
+#include <cstddef>
+#include <cstdint>
+#include <exception>
+#include <functional>
+#include <iterator>
+#include <list>
+#include <type_traits>
+#include <utility>
+
+namespace hpx { namespace parallel { inline namespace v1 {
+
+    ///////////////////////////////////////////////////////////////////////////
+    // sort
+    namespace detail {
+
+        /// \cond NOINTERNAL
+
+        //////////////////////////////////////////////////////////////////////
+        ///
+        /// \brief Obtain the position of the most significant bit set in N
+        ///
+        /// \param num : Number to examine
+        ///
+        /// \return Position of the first bit set
+        ///
+        inline constexpr unsigned nbits64(std::uint64_t num) noexcept
+        {
+            unsigned nb = 0;
+            if (num >= (1ull << 32))
+            {
+                nb += 32;
+                num = (num >> 32);
+            }
+            if (num >= (1ull << 16))
+            {
+                nb += 16;
+                num = (num >> 16);
+            }
+            if (num >= (1ull << 8))
+            {
+                nb += 8;
+                num = (num >> 8);
+            }
+            if (num >= (1ull << 4))
+            {
+                nb += 4;
+                num = (num >> 4);
+            }
+            if (num >= (1ull << 2))
+            {
+                nb += 2;
+                num = (num >> 2);
+            }
+            if (num >= (1ull << 1))
+            {
+                nb += 12;
```

Comment:
Shouldn't this be `1`?

Reply:
That's a very good catch! It sure looks like it.

Reply:
Will fix (straight to master). Thanks!